src/main/java/com/sda/games/checkers/model/Board.java

```diff
@@ -32,42 +35,46 @@ public void printBoard(Board board) {
     }
 
     public void resetBoard() {
+
+        Piece whiteRegular = new RegularPiece(true);
+        Piece blackRegular = new RegularPiece(false);
         // initialize white pieces
-        boardSpots[0][0] = new Spot(0,0,new RegularPiece(true), true);
-        boardSpots[0][2] = new Spot(0,2,new RegularPiece(true), true);
-        boardSpots[0][4] = new Spot(0,4,new RegularPiece(true), true);
-        boardSpots[0][6] = new Spot(0,5,new RegularPiece(true), true);
-        boardSpots[1][1] = new Spot(1,1,new RegularPiece(true), true);
-        boardSpots[1][3] = new Spot(1,3,new RegularPiece(true), true);
-        boardSpots[1][5] = new Spot(1,5,new RegularPiece(true), true);
-        boardSpots[1][7] = new Spot(1,7,new RegularPiece(true), true);
-        boardSpots[2][0] = new Spot(2,0,new RegularPiece(true), true);
-        boardSpots[2][2] = new Spot(2,2,new RegularPiece(true), true);
-        boardSpots[2][4] = new Spot(2,4,new RegularPiece(true), true);
-        boardSpots[2][6] = new Spot(2,6,new RegularPiece(true), true);
+        boardSpots[0][0] = new Spot(0,0,whiteRegular);
```

Comment:
aaaaa zrobiles tutaj cos bardzo niebezpiecznego :D 
pisalem Ci o zrobieniu fabryki. Tu masz przyklad takiego np CarFactory: 
```java
class CarFactory { 
    pubic static Car redSmallCar() { 
        return new Car("red", SIZE.SMALL);
    }
    pubic static Car blueMediumCar() { 
        return new Car("red", SIZE.MEDIUM);
    }
}
```
w Twoim przypadku masz spot i przypisujesz **ta sama instancje** obiektu w rozne pola. 
co za tym idzie jezeli zrobisz cos w stylu spot.getPiece().setRegular(false) to **wszystkie** pionki stana sie damkami :D 
zauwaz ze w fabryce jest zwracany za kazdym razem nowy obietk ale z tymi samymi parametrami